File: sorting/insertionSort.py
# Insertion sort is a simple sorting algorithm that works the way we sort playing cards in our hands.

# Algo

# 0       1           2           3           4
# key = 11

def insertionSort(arr):
  
    # Traverse through 1 to len(arr)
    for i in range(1, len(arr)):
  
        key = arr[i]
  
        # Move elements of arr[0..i-1], that are
        # greater than key, to one position ahead
        # of their current position
        j = i-1
        while j >=0 and key < arr[j] :
                arr[j+1] = arr[j]
                j -= 1
        arr[j+1] = key
  
arr = [12, 11, 13, 5, 6]
insertionSort(arr)
print ("Sorted array is:")
for i in range(len(arr)):
    print ("%d" %arr[i])

File: sorting/insertionSort.js

function sortD(data){
for (let i = 0; i <= data.length; i++) {
    let j = i;
    while (j > 0 && data[i] < data[j - 1])
      j--;
    let tmp = data[i];
    for (let k = i; k > j; k--){
      data[k] = data[k - 1];
    data[j] = tmp;
    }
  }
  return data;

}

const val = [2, 3, 1, 4];
const rt = sortD(val);
console.log(rt, 'rt')

// node sorting/insertionSort.js